code/src/reducers/todos.js

```diff
@@ -0,0 +1,42 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+const todos = createSlice({
+  name: 'todos',
+  initialState: {
+    items: [
+    ]
+  },
+  reducers: {
+    toggleTodoItem: (store, action) => {
+      console.log(store);
+      console.log(action);
+      store.items.forEach((item) => {
+        if (item.id === action.payload) {
+          item.isDone = !item.isDone
+        }
+      });
+    },
+    deleteTodoItem: (store, action) => {
+      console.log(store);
+      console.log(action);
+      // mutability
+      store.items.splice(action.payload, 1); // deleting one item
+      // immutability
+      /* const updatedItems = store.items.filter((item) => {
+        return store.items.indexOf(item) !== action.payload
+      });
+      store.items = updatedItems */
+    },
+    addItem: (store, action) => {
+      console.log(store);
+      console.log(action);
+      store.items.push(action.payload)
+      /* store.items = [...store.items, action.payload] */
+    },
+    setAllItems: (store, action) => {
+      store.items = action.payload;
+    }
```

Comment:
If you would like to implement a clear all button, it could look like this: 
`clearAllTasks: (store) => {
      store.items = [];
    },`